src/components/data/DataNavigation.js

```diff
@@ -0,0 +1,460 @@
+        textOverflow: "ellipsis",
+        whiteSpace: "nowrap",
+        overflow: "hidden",
+        display: "inline-block",
+        maxWidth: 90,
+    },
+}));
+
+/**
+ * Splits a path into an array of path items.
+ * @example
+ * // returns [analyses,wordcount,logs]
+ * getPathItems("/analyses/wordcount/logs");
+ * @param {string }relativePath - relative path to CyVerse Data Store e.g: /analyses/wordcount/logs
+ * @returns {string[]} - path items as a string array  e.g: [analyses,wordcount,logs]
+ */
+function getPathItems(relativePath) {
+    let pathItems = [];
+    pathItems = relativePath.split("/").slice(1);
+    return pathItems;
+}
+
+/**
+ * Computes the path for routing.
+ *  * @example
+ * // returns "/data?path=/iplant/home/ipctest/analyses"
+ * getPathItems("/iplant/home/ipctest","/analyses/wordcount/logs", 0);
+ * @param {string }root - CyVerse Data Store root path e.g: /iplant/home/ipctest
+ * @param {string} relativePath - relative path to CyVerse Data Store e.g: /analyses/wordcount/logs
+ * @param {number} selectedPathItemIndex - index of the selected path item
+ * @returns {string} - path to be used by the nextjs router
+ */
+function pathToRoute(root, relativePath, selectedPathItemIndex) {
+    const relativePathItems = getPathItems(relativePath);
+    const reducer = (accumulator, currentVal, idx) => {
+        if (idx <= selectedPathItemIndex) {
+            return accumulator + "/" + currentVal;
+        }
+        return accumulator;
+    };
+    const routerPath = relativePathItems.reduce(reducer);
+    console.log(
+        "reduced path=>" +
+            routerPath +
+            " path index=>" +
+            selectedPathItemIndex +
+            " relative path=>" +
+            relativePathItems.length
+    );
+
+    return "/" + NavigationConstants.DATA + `?path=${root}/${routerPath}`;
+}
+
+/**
+ * Get relative path to CyVerse Data Store path.
+ *  * @example
+ * // returns "/analyses/wordcount/logs"
+ * getRelativePath("/iplant/home/ipctest/analyses/wordcount/logs","/iplant/home/ipctest",
+ * "/iplant/trash/home/de-irods/ipctest");
+ * @param {string} path -  CyVerse Data Store path
+ * @param {string} userHomePath - Users home folder path
+ * @param {string} userTrashPath - Users trash folder path
+ * @returns {string} relativePath - relative path to CyVerse Data Store e.g: /analyses/wordcount/logs
+ */
+function getRelativePath(path, userHomePath, userTrashPath) {
+    let relativePath = "";
+    if (path.indexOf(userHomePath) !== -1) {
+        relativePath = path.replace(userHomePath, "");
+    } else if (path.indexOf(userTrashPath) !== -1) {
+        relativePath = path.replace(userTrashPath, "");
+    } else if (path.indexOf(constants.COMMUNITY_DATA_PATH) !== -1) {
+        relativePath = path.replace(constants.COMMUNITY_DATA_PATH, "");
+    } else if (path.indexOf(constants.SHARED_WITH_ME_PATH) !== -1) {
+        relativePath = path.replace(constants.SHARED_WITH_ME_PATH, "");
+    }
```

Comment:
It might be safer to use `path.startsWith` instead of `indexOf`.
And the string passed to `startsWith` should probably end with a `/` to be extra safe.